vendor/bat-native-ledger/src/bat/ledger/internal/gemini/gemini_transfer.cc

```diff
@@ -60,5 +65,59 @@ void GeminiTransfer::OnCreateTransaction(const type::Result result,
   callback(type::Result::LEDGER_OK, id);
 }
 
+void GeminiTransfer::StartTransactionStatusTimer(
+    const std::string& id,
+    const int attempts,
+    client::TransactionCallback callback) {
+  auto new_attempts = attempts + 1;
+  auto mins = 3 * new_attempts;
+  base::TimeDelta delay = base::TimeDelta::FromMinutes(mins);
+  BLOG(1, "Will fetch transaction status after " << mins << " minutes");
+  retry_timer_.Start(
+      FROM_HERE, delay,
+      base::BindOnce(&GeminiTransfer::FetchTransactionStatus,
+                     base::Unretained(this), id, new_attempts, callback));
+}
+
+void GeminiTransfer::FetchTransactionStatus(
+    const std::string& id,
+    const int attempts,
+    client::TransactionCallback callback) {
+  retry_timer_.Stop();
```

Comment:
Is this necessary? Since this is a callback for a one-shot timer it seems like it will always be stopped here.

Reply:
fixed!